File: test-scripts/send_test_email.go
package main

import (
	"flag"
	"fmt"
	"log"
	"net/smtp"
	"os"
)

func main() {
	// Configure logging
	log.SetFlags(log.Ldate | log.Ltime)
	log.SetPrefix("[test-email] ")

	// Parse command line arguments
	host := flag.String("host", "localhost", "SMTP server host")
	port := flag.Int("port", 25, "SMTP server port")
	fromAddr := flag.String("from", "sender@example.com", "From email address")
	toAddr := flag.String("to", "foo@localhost.localdomain", "To email address")
	subject := flag.String("subject", "Test Subject Word1 Word2", "Email subject")
	body := flag.String("body", "This is a test email body.", "Email body")
	flag.Parse()

	// Build email message
	msg := fmt.Sprintf("From: %s\r\n"+
		"To: %s\r\n"+
		"Subject: %s\r\n"+
		"\r\n"+
		"%s\r\n", *fromAddr, *toAddr, *subject, *body)

	// Prepare server address
	addr := fmt.Sprintf("%s:%d", *host, *port)
	log.Printf("Attempting to connect to %s...", addr)

	// Create SMTP client
	client, err := smtp.Dial(addr)
	if err != nil {
		if os.IsTimeout(err) {
			log.Fatalf("Connection timeout - Is the SMTP server running on %s?", addr)
		}
		log.Fatalf("Failed to connect: %v", err)
	}
	defer client.Close()
	log.Println("Connected to server")

	// Set sender and recipient
	if err := client.Mail(*fromAddr); err != nil {
		log.Fatalf("Failed to set sender: %v", err)
	}
	if err := client.Rcpt(*toAddr); err != nil {
		log.Fatalf("Failed to set recipient: %v", err)
	}

	// Send the email body
	log.Println("Attempting to send message...")
	writer, err := client.Data()
	if err != nil {
		log.Fatalf("Failed to start data transaction: %v", err)
	}

	_, err = writer.Write([]byte(msg))
	if err != nil {
		log.Fatalf("Failed to write message: %v", err)
	}

	err = writer.Close()
	if err != nil {
		log.Fatalf("Failed to close data transaction: %v", err)
	}

	// Quit the connection
	err = client.Quit()
	if err != nil {
		log.Printf("Warning: Failed to close connection cleanly: %v", err)
	}

	// Print success message
	log.Println("\nEmail sent successfully!")
	log.Printf("From: %s", *fromAddr)
	log.Printf("To: %s", *toAddr)
	log.Printf("Subject: %s", *subject)
	log.Printf("Body: %s", *body)
}

File: test-scripts/send_test_email.py
#!/usr/bin/env python3
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart
import argparse


def send_email(smtp_host, smtp_port, from_addr, to_addr, subject, body):
    # Create the email message
    msg = MIMEMultipart()
    msg["From"] = from_addr
    msg["To"] = to_addr
    msg["Subject"] = subject

    # Add body to email
    msg.attach(MIMEText(body, "plain"))

    try:
        print(f"Attempting to connect to {smtp_host}:{smtp_port}...")
        # Create SMTP connection with debugging
        with smtplib.SMTP(smtp_host, smtp_port, timeout=10) as server:
            server.set_debuglevel(1)  # Enable debug output
            print("Connected to server")

            # Send email
            print("Attempting to send message...")
            server.send_message(msg)
            print(f"\nEmail sent successfully!")
            print(f"From: {from_addr}")
            print(f"To: {to_addr}")
            print(f"Subject: {subject}")
            print(f"Body: {body}")
    except ConnectionRefusedError:
        print(
            f"Connection refused - Is the SMTP server running on {smtp_host}:{smtp_port}?"
        )
    except smtplib.SMTPServerDisconnected as e:
        print(f"Server disconnected: {e}")
        print("This might happen if:")
        print("1. The server doesn't accept the connection")
        print("2. The server requires authentication")
        print("3. The server closed the connection unexpectedly")
    except Exception as e:
        print(f"Failed to send email: {e}")
        print(f"Error type: {type(e).__name__}")


def main():
    parser = argparse.ArgumentParser(
        description="Send a test email to a local SMTP server"
    )
    parser.add_argument(
        "--host", default="localhost", help="SMTP server host (default: localhost)"
    )
    parser.add_argument(
        "--port", type=int, default=25, help="SMTP server port (default: 25)"
    )
    parser.add_argument(
        "--from",
        dest="from_addr",
        default="sender@example.com",
        help="From email address",
    )
    parser.add_argument(
        "--to", dest="to_addr", default="recipient@example.com", help="To email address"
    )
    parser.add_argument(
        "--subject", default="Test Subject Word1 Word2", help="Email subject"
    )
    parser.add_argument(
        "--body", default="This is a test email body.", help="Email body"
    )

    args = parser.parse_args()

    send_email(
        args.host, args.port, args.from_addr, args.to_addr, args.subject, args.body
    )


if __name__ == "__main__":
    main()
